src/Compilers/CSharp/Test/Semantic/Semantics/MethodTypeInferenceTests.cs

```diff
@@ -633,6 +635,155 @@ static void Main(string[] args)
+                Assert.Null(model.GetDeclaredSymbol(discardOut));
+                var discardSymbol = (IDiscardSymbol)model.GetSymbolInfo(discardOut).Symbol;
+                Assert.Equal("System.String", discardSymbol.Type.ToTestDisplayString());
+                Assert.Equal("System.String", model.GetTypeInfo(discardOut).Type.ToTestDisplayString());
+            });
+
+
+            // out T _, out var _, out T? _
+            ForEach(GetDiscardDesignations(tree), (DiscardDesignationSyntax discardDecl) =>
+            {
+                Assert.Null(model.GetDeclaredSymbol(discardDecl));
+                Assert.Null(model.GetTypeInfo(discardDecl).Type);
+                Assert.Null(model.GetSymbolInfo(discardDecl).Symbol);
+                var declaration = (DeclarationExpressionSyntax)discardDecl.Parent;
+                Assert.Equal("System.String", model.GetTypeInfo(declaration).Type.ToTestDisplayString());
+                Assert.Null(model.GetSymbolInfo(declaration).Symbol);
+            });
+        }
+
+        [Fact]
+        public void Bug50782_1() {
+            string source = @"
+using System.Diagnostics.CodeAnalysis;
+#nullable enable
+
+interface IOperation<T> { }
+
+public class StringOperation : IOperation<string?> { }
+
+public class C {   
+    static void Main() {
+        TestA(new StringOperation(), out string? discardA);
+        TestA(new StringOperation(), out string? _);
+        TestA(new StringOperation(), out var _);
+        TestA(new StringOperation(), out _);
+        
+        TestB(new StringOperation(), out string? discardB);
+        TestB(new StringOperation(), out string? _);
+        TestB(new StringOperation(), out var _);
+        TestB(new StringOperation(), out _);
+        
+        TestC<string?>(out string? discardC);
+        TestC<string?>(out string? _);
+        TestC<string?>(out var _);
+        TestC<string?>(out _);
+        
+        TestD<string?>(out string? discardD);
+        TestD<string?>(out string? _);
+        TestD<string?>(out var _);
+        TestD<string?>(out _);
+        
+        TestE(out string? discardE);
+        TestE(out var descardEVar);
+        TestE(out string? _);
+        TestE(out var _);
+        TestE(out _);
+        
+        TestF(out string? discardF);
+        TestF(out string? _);
+    }
+   
+    static void TestA<T>(IOperation<T> operation, [MaybeNull] out T result) => result = default;
+    static void TestB<T>(IOperation<T> operation, out T result) => result = default;
+    
+    static void TestC<T>([MaybeNull] out T result) => result = default;
+    static void TestD<T>(out T result) => result = default;
+
+    static void TestE([MaybeNull] out string result) => result = default;
+    static void TestF(out string result) => result = """";
+}
+";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+            comp.VerifyDiagnostics(new[] {
```

Comment:
This should look more like this:
```
            compilation.VerifyDiagnostics(
                // (34,13): warning CS8602: Dereference of a possibly null reference.
                //             ns /*T:string?*/ .ToString(); // 3
                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "ns").WithLocation(34, 13),
                // (44,13): warning CS8602: Dereference of a possibly null reference.
                //             ns1 /*T:string?*/ .ToString(); // 4
                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "ns1").WithLocation(44, 13)
                );
```

To achieve this, run the test in a way that `VerifyDiagnostics` fails, then copy&paste from the test output.